connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/arrow/ScalaCollectionUtils.scala

```diff
@@ -34,4 +34,8 @@ private[arrow] object ScalaCollectionUtils {
   def wrap[T](array: AnyRef): mutable.ArraySeq[T] = {
     mutable.ArraySeq.make(array.asInstanceOf[Array[T]])
   }
+
+  def toImmutableArraySeq[T](array: AnyRef): immutable.ArraySeq[T] = {
```

Comment:
nit: Perhaps we can directly use the implicit conversion defined in `ArrayImplicits` instead of add this new function, which is now a more generic conversion way in Spark code.

Reply:
Done